main.tf

```diff
@@ -99,6 +99,28 @@ resource "aws_nat_gateway" "natgw" {
   depends_on = ["aws_internet_gateway.mod"]
 }
 
+data "aws_vpc_endpoint_service" "s3" {
+  service = "s3"
+}
+
+resource "aws_vpc_endpoint" "ep" {
+  vpc_id       = "${aws_vpc.mod.id}"
+  service_name = "${data.aws_vpc_endpoint_service.s3.service_name}"
+  count        = "${var.enable_s3_endpoint ? 1 : 0}"
```

Comment:
Part ` ? 1 : 0` is not needed. `count        = "${var.enable_s3_endpoint}"` - is enough